### PR TITLE
[prometheus-blackbox-exporter] Added support of sidecars and extraVolumes

### PR DESCRIPTION
**Pull request by pburgisser**
Signed-off-by: Philippe Bürgisser <philippe.burgisser@camptocamp.com>

#### What this PR does / why we need it:
- Adding support of multiple container in the same pod (side car, like oAuth-proxy)
- Adding support of multiple secret (necessary for oauth)
- Adding support of annotation on service, service account
- Adding support of authenticated Service Monitor

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-blackbox-exporter/templates/deployment.yaml [hunk 1]
```diff
@@ -87,7 +90,7 @@ spec:
           resources:
 {{ toYaml .Values.resources | indent 12 }}
           ports:
-            - containerPort: {{ .Values.service.port }}
+            - containerPort: {{ .Values.port }}
```
Comment (monotek) [inline charts/prometheus-blackbox-exporter/templates/deployment.yaml:93]: Why the change?
Now port is configurable twice?

File: charts/prometheus-blackbox-exporter/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -16,7 +16,15 @@ metadata:
 spec:
   endpoints:
   - port: http
-    scheme: http
+    {{- if .Values.serviceMonitor.scheme }}
+    scheme: {{ .Values.serviceMonitor.scheme }}
```
Comment (monotek) [inline charts/prometheus-blackbox-exporter/templates/servicemonitor.yaml:20]: this breaks current behaviour as the value would change.
imho this would make this a breaking change.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-blackbox-exporter/templates/deployment.yaml (hunk 1 above)
Comment (pburgisser) [reply charts/prometheus-blackbox-exporter/templates/deployment.yaml:93]: If you add a oAuthProxy in front, you need to differentiate the service port and the container port, so there is not direct access to the blackbox.
In Prometheus-pushgateway, the port if hard coded [1], maybe we can hard code it too ?

[1] https://github.com/prometheus-community/helm-charts/blob/a72459c5a1a5672c0bcd812a5a078ec4517693d2/charts/prometheus-pushgateway/templates/deployment.yaml#L40

File: charts/prometheus-blackbox-exporter/templates/servicemonitor.yaml (hunk 2 above)
Comment (pburgisser) [reply charts/prometheus-blackbox-exporter/templates/servicemonitor.yaml]: Good catch !
Changed the default values to http so I can keep changing the scheme in case I'm doing authenticated prometheus (with oAuth proxy on OCP)

File: charts/prometheus-blackbox-exporter/templates/deployment.yaml (hunk 1 above)
Comment (monotek) [reply charts/prometheus-blackbox-exporter/templates/deployment.yaml:93]: Why don't you add a targetPort key to the service and make it configurable by a variable like `targetPort: {{ .Values.service.targetPort }}`?
With this you could set the proxy to be accessed by the service instead of the alertmanager container.

Comment (pburgisser) [reply charts/prometheus-blackbox-exporter/templates/deployment.yaml]: OK, I did as you suggested. I hope that works for you now.

File: charts/prometheus-blackbox-exporter/values.yaml [hunk 3]
```diff
@@ -96,7 +126,7 @@ service:
   labels: {}
   type: ClusterIP
   port: 9115
-
+  targetPort: 9115
```
Comment (monotek) [inline charts/prometheus-blackbox-exporter/values.yaml:129]: Please add blank line below again

File: charts/prometheus-blackbox-exporter/values.yaml [hunk 4]
```diff
@@ -39,6 +68,7 @@ runAsUser: 1000
 readOnlyRootFilesystem: true
 runAsNonRoot: true
 
+
```
Comment (monotek) [inline charts/prometheus-blackbox-exporter/values.yaml:71]: Please remove double blank line here.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-blackbox-exporter/values.yaml [hunk 5]
```diff
@@ -13,6 +13,35 @@ podDisruptionBudget: {}
...
 
+extraVolumes: []
+  # - name: secret-blackbox-oauth-htpasswd
+  #   secret:
+  #     defaultMode: 420
+  #     secretName: blackbox-oauth-htpasswd
+  # - name: storage-volume
+  #   persistentVolumeClaim:
+  #     claimName: example
+extraContainers: []
```
Comment (monotek) [inline charts/prometheus-blackbox-exporter/values.yaml:24]: Add blank line above please

Comment (pburgisser) [reply charts/prometheus-blackbox-exporter/values.yaml:25]: There you go

**Review by monotek — APPROVED** [review]
